mailbox/store/src/main/java/org/apache/james/mailbox/store/StoreMailboxManager.java

```diff
@@ -418,6 +418,36 @@ protected StoreMessageManager createMessageManager(Mailbox mailbox, MailboxSessi
     }
 
     @Override
+    public org.apache.james.mailbox.MessageManager getMailbox(MailboxId mailboxId, MailboxSession session)
+            throws MailboxException {
+        MailboxMapper mapper = mailboxSessionMapperFactory.getMailboxMapper(session);
+        Mailbox mailboxRow = mapper.findMailboxById(mailboxId);
+
+        if (mailboxRow == null) {
+            session.getLog().info("Mailbox '" + mailboxId.serialize() + "' not found.");
+            throw new MailboxNotFoundException(mailboxId.serialize());
+        }
+        
+        if (! belongsToCurrentUser(mailboxRow, session)) {
+            session.getLog().info("Mailbox '" + mailboxId.serialize() + "' does not belong to user '" + session.getUser() + "' but to '" + mailboxRow.getUser());
+            throw new MailboxNotFoundException(mailboxId.serialize());
+        }
+
+        session.getLog().debug("Loaded mailbox " + mailboxId.serialize());
+
+        StoreMessageManager m = createMessageManager(mailboxRow, session);
+        m.setFetchBatchSize(fetchBatchSize);
+        return m;
+    }
+
+    private boolean belongsToCurrentUser(Mailbox mailbox, MailboxSession session) {
+        if (session.getUser() == null || session.getUser().getUserName() == null) {
+            return mailbox.getUser() == null;
+        }
+        return session.getUser().getUserName().equals(mailbox.getUser());
```

Comment:
Mail address are not case sensitive. User names directly comes from mail addresses.
However user are defined from UserRepository, which are case insentive.